pkg/expr/convert_to_long_test.go

```diff
@@ -46,3 +47,189 @@ func TestConvertNumericWideToLong(t *testing.T) {
 		require.FailNowf(t, "Result mismatch (-want +got):%s\n", diff)
 	}
 }
+
+func TestConvertTimeSeriesMultiToLong(t *testing.T) {
+	t.Run("UnalignedTimestampsWithNamedFields", func(t *testing.T) {
+		tLGA := []time.Time{
+			time.UnixMilli(1664901845976),
+			time.UnixMilli(1664902845976),
+		}
+		valLGA := []float64{3, 5}
+
+		tMIA := []time.Time{
+			time.UnixMilli(1664901855976),
+			time.UnixMilli(1664902455976),
+			time.UnixMilli(1664902855976),
+		}
+		valMIA := []float64{6, 7, 9}
+
+		input := data.Frames{
+			data.NewFrame("frameLGA",
+				data.NewField("t", nil, tLGA),
+				data.NewField("slothCount", data.Labels{"city": "LGA"}, valLGA),
+			),
+			data.NewFrame("frameMIA",
+				data.NewField("t", nil, tMIA),
+				data.NewField("slothCount", data.Labels{"city": "MIA"}, valMIA),
+			),
+		}
+
+		expected := data.NewFrame("time_series_long",
+			data.NewField("t", nil, []time.Time{
+				tLGA[0],
+				tMIA[0],
+				tMIA[1],
+				tLGA[1],
+				tMIA[2],
+			}),
+			data.NewField("slothCount", nil, []float64{
+				3, 6, 7, 5, 9,
+			}),
+			data.NewField("city", nil, []string{
+				"LGA", "MIA", "MIA", "LGA", "MIA",
+			}),
+		)
+		expected.Meta = &data.FrameMeta{Type: data.FrameTypeTimeSeriesLong}
+
+		output, err := convertTimeSeriesMultiToTimeSeriesLong(input)
+		require.NoError(t, err)
+		require.Len(t, output, 1)
+
+		if diff := cmp.Diff(expected, output[0], data.FrameTestCompareOptions()...); diff != "" {
+			require.FailNowf(t, "UnalignedTimestampsWithNamedFields mismatch (-want +got):\n%s", diff)
+		}
+	})
+
+	t.Run("LabelOverwrite", func(t *testing.T) {
+		times := []time.Time{
+			time.Unix(0, 0),
+			time.Unix(15, 0),
+		}
+
+		input := data.Frames{
+			data.NewFrame("frame1",
+				data.NewField("ts", nil, times),
+				data.NewField("val", data.Labels{"namespace": "foo"}, []float64{1, 2}),
+			),
+			data.NewFrame("frame2",
+				data.NewField("ts", nil, times),
+				data.NewField("val", data.Labels{"namespace": "bar"}, []float64{3, 4}),
+			),
+		}
+
+		expected := data.NewFrame("time_series_long",
+			data.NewField("ts", nil, []time.Time{
+				times[0], // 0s - foo
+				times[0], // 0s - bar
+				times[1], // 15s - foo
+				times[1], // 15s - bar
+			}),
+			data.NewField("val", nil, []float64{
+				1, 3, 2, 4,
+			}),
+			data.NewField("namespace", nil, []string{
+				"foo", "bar", "foo", "bar",
```

Comment:
I notice these are not in alphabetical order.
Could we sort alphabetically, instead of by label fingerprint? I think it's more user friendly. No idea if it's harder - I could imagine it might be.